mvrx-navigation/build.gradle

```diff
@@ -0,0 +1,135 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+apply plugin: 'jacoco'
+apply plugin: 'com.vanniktech.maven.publish'
+
+android {
+    compileSdkVersion versions.compileSdk
+    buildToolsVersion versions.buildTools
+    resourcePrefix "mvrx_"
+
+    defaultConfig {
+        minSdkVersion versions.minSdk
+        targetSdkVersion versions.targetSdk
+        consumerProguardFiles "proguard-rules.pro"
+    }
+
+    lintOptions {
+        abortOnError true
+        textReport true
+        textOutput "stdout"
+    }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_1_8.toString()
+    }
+
+    sourceSets {
+        main.java.srcDirs += 'src/main/kotlin'
+        test.java.srcDirs += 'src/test/kotlin'
+        test.res.srcDirs += 'src/test/res'
+    }
+
+    testOptions {
+        unitTests {
+            includeAndroidResources = true
+            all {
+                jacoco {
+                    includeNoLocationClasses = true
+                }
+            }
+        }
+    }
+}
+
+androidExtensions {
+    experimental = true
+}
+
+jacoco {
+    toolVersion = "0.8.2"
+}
+
+dependencies {
+    implementation project(":mvrx")
+    implementation project(":testing")
+
+    implementation libraries.lifecycleCommonJava8
+    implementation libraries.kotlin
+
+    testImplementation testLibraries.mockito
+    testImplementation testLibraries.junit
+    testImplementation testLibraries.roboeletric
+    testImplementation libraries.rxAndroid
+}
+
+// module specific dependencies
+dependencies {
+    def nav_version = "2.3.0-alpha04"
+    implementation "androidx.navigation:navigation-fragment-ktx:$nav_version"
+    implementation "androidx.navigation:navigation-ui-ktx:$nav_version"
+
+    def fragment_version = "1.2.4"
+    debugImplementation "androidx.fragment:fragment:$fragment_version"
+    debugImplementation "androidx.fragment:fragment-ktx:$fragment_version"
+    debugImplementation "androidx.fragment:fragment-testing:$fragment_version"
+    testImplementation "androidx.navigation:navigation-testing:$nav_version"
+}
+
+// Ensure custom source sets are included:
+// https://www.virag.si/2015/01/publishing-gradle-android-library-to-jcenter/
+task sourcesJar(type: Jar) {
+    from android.sourceSets.main.java.srcDirs
+    classifier = 'sources'
+}
+
+task javadoc(type: Javadoc) {
+    source = android.sourceSets.main.java.srcDirs
+    classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+artifacts {
+    archives javadocJar
+    archives sourcesJar
+}
+
+tasks.withType(Javadoc).all {
+    enabled = false
+}
+
+task coverage(type: JacocoReport, dependsOn: "testDebugUnitTest") {
```

Comment:
@marukami *absolute n00b Gradle question:*

Since this task is also present at [`mvrx`](https://github.com/airbnb/MvRx/blob/master/mvrx/build.gradle), could it be moved to the root [`build.gradle`](https://github.com/airbnb/MvRx/blob/master/build.gradle) file?

Reply:
Yep, the best way is to create a named task in buildSrc and then loop over subprojects and add the task to every module. The tricky thing here is that we have example modules mixed in. Given we only have a few modules a simple custom plugin would reduce the most code per module.